models/actions/task_step.go

```diff
@@ -0,0 +1,41 @@
+// Copyright 2022 The Gitea Authors. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+package actions
+
+import (
+	"context"
+	"time"
+
+	"code.gitea.io/gitea/models/db"
+	"code.gitea.io/gitea/modules/timeutil"
+)
+
+// ActionTaskStep represents a step of ActionTask
+type ActionTaskStep struct {
+	ID        int64
+	Name      string `xorm:"VARCHAR(255)"`
+	TaskID    int64  `xorm:"index unique(task_index)"`
+	Index     int64  `xorm:"index unique(task_index)"`
+	RepoID    int64  `xorm:"index"`
+	Status    Status `xorm:"index"`
+	LogIndex  int64
+	LogLength int64
+	Started   timeutil.TimeStamp
+	Stopped   timeutil.TimeStamp
+	Created   timeutil.TimeStamp `xorm:"created"`
+	Updated   timeutil.TimeStamp `xorm:"updated"`
+}
+
+func (step *ActionTaskStep) Duration() time.Duration {
+	return calculateDuration(step.Started, step.Stopped, step.Status)
+}
+
+func init() {
+	db.RegisterModel(new(ActionTaskStep))
+}
+
+func GetTaskStepsByTaskID(ctx context.Context, taskID int64) ([]*ActionTaskStep, error) {
+	var steps []*ActionTaskStep
+	return steps, db.GetEngine(ctx).Where("task_id=?", taskID).OrderBy("index").Find(&steps)
```

Comment:
I had to escape the sqlite keyword `index`, more details here https://github.com/go-gitea/gitea/pull/21937#discussion_r1063365091 and https://www.sqlite.org/lang_keywords.html (this lists index as a keyword for the sqlite sql dialect)

```suggestion
	return steps, db.GetEngine(ctx).Where("task_id=?", taskID).OrderBy("\"index\"").Find(&steps)
```

_Otherwise avoid sqlite keywords as table column names and rename index again_

This prevents viewing jobs of a repository and probably other things with a sqlite db.

Reply:
Rewirted it to ``OrderBy("`index` ASC")``, it should works now.

There're lots of models that have a field named `Index`, so I think there's no need to rename it.

Reply:
There are only three other models (`CommitStatus`, `Issue`, `PullRequest`) which use a `Index` field and only `CommitStatus` uses it in a sql query. You should use `.Asc("index")` which will automatically escape the field name.

Reply:
For my use cases would non standard sql grave accents work too (sqlite/mysql). However I would prefer `.Asc("index")` like mentioned above.